### ISSUE TITLE
Does not implement Send + Sync

### ISSUE THREAD

**Issue by Tails** [SolraBizna/arcow#1]
Arcow is exactly what I was looking for, thank you!

However, when using it in a data structure with async fn methods, it complains that Arcow is not Send nor Sync.

Is that intentional? 

```rust
/// simple cache that retires objects that have not been in use for
/// a specific amount of time
#[derive(Derivative, Clone, Debug)]
pub struct MemoryUnionStorage<K: StorageAddress, V: UnionDeps> {
    objects: Arc<DashMap<K, Arcow<V>>>,
}
```

```bash
error[E0277]: `NonNull<arcow::ArcowInner<V>>` cannot be sent between threads safely
   --> lib/graph/graph2/src/providers/heap.rs:56:47
    |
56  | impl <K: StorageAddress, V: UnionDeps + Sync> StorageProvider for MemoryUnionStorage<K, V> where V: ToStorageAddress<K> + DefaultToStorag...
    |                                               ^^^^^^^^^^^^^^^ `NonNull<arcow::ArcowInner<V>>` cannot be sent between threads safely
    |
    = help: within `Arcow<V>`, the trait `std::marker::Send` is not implemented for `NonNull<arcow::ArcowInner<V>>`
    = note: required because it appears within the type `Arcow<V>`
    = note: required for `dashmap::util::SharedValue<Arcow<V>>` to implement `std::marker::Send`
    = note: required because it appears within the type `(K, dashmap::util::SharedValue<Arcow<V>>)`
    = note: required for `hashbrown::raw::RawTable<(K, dashmap::util::SharedValue<Arcow<V>>)>` to implement `std::marker::Send`
    = note: required because it appears within the type `hashbrown::map::HashMap<K, dashmap::util::SharedValue<Arcow<V>>, RandomState>`
    = note: required for `lock_api::rwlock::RwLock<dashmap::lock::RawRwLock, hashbrown::map::HashMap<K, dashmap::util::SharedValue<Arcow<V>>, RandomState>>` to implement `Sync`
    = note: required because it appears within the type `[lock_api::rwlock::RwLock<dashmap::lock::RawRwLock, hashbrown::map::HashMap<K, dashmap::util::SharedValue<Arcow<V>>, RandomState>>]`
    = note: required for `std::ptr::Unique<[lock_api::rwlock::RwLock<dashmap::lock::RawRwLock, hashbrown::map::HashMap<K, dashmap::util::SharedValue<Arcow<V>>, RandomState>>]>` to implement `Sync`
    = note: required because it appears within the type `Box<[lock_api::rwlock::RwLock<dashmap::lock::RawRwLock, hashbrown::map::HashMap<K, dashmap::util::SharedValue<Arcow<V>>, RandomState>>]>`
    = note: required because it appears within the type `DashMap<K, Arcow<V>>`
    = note: required for `std::sync::Arc<DashMap<K, Arcow<V>>>` to implement `Sync`

```

**Comment by SolraBizna** [comment]
It should at least be `Send`, probably also `Sync`. I'll look into it.

**Comment by SolraBizna** [comment]
I've modified it to implement `Send` and `Sync` if the associated type is `Send + Sync`, same as with standard `Arc`. I don't know if that will be enough for your use case, but that's the most I'm sure of at the moment. I think I can make a stronger guarantee but I'll need to take some time to think about it.

Try the latest revision, which I'm not yet ready to publish on `crates.io`:

```toml
[dependencies]
arcow = {git = "https://github.com/SolraBizna/arcow", rev = "005f75ea333bcdea9bf1ceb87d51e38edbbd194c"}
```

and let me know if this solves your problem.

**Comment by Tails** [comment]
It works! 

It would be nice if PartialEq would also be implemented where T: PartialEq<T>, but that can be worked around with custom newtypes.

Thank you for sharing Arcow!

**Comment by SolraBizna** [comment]
That's a really good idea.

I've now published 1.1.0, which contains the `Send+Sync` change, and also implements `PartialEq`, `Eq`, `PartialOrd`, and `Ord` whenever the underlying type makes this possible.

Enjoy :)
